### PR TITLE
[Salesforce] Add pagination support

### PR DESCRIPTION
**Pull request by kush-elastic**
## What does this PR do?
In the current Salesforce Module in filebeat, we don’t have pagination support for all four file-sets (login-rest, logout-rest, apex-rest,  setupaudittrail-rest). So if in any case, we have a response body with more than 2000 records, the next page is never called and the data will not be fetched.

Solution:
You can request the next batch of records using the provided URL with your instance and session information, and repeat until all records have been retrieved. These requests use `nextRecordsUrl` and don’t include any parameters. The final batch of records doesn’t have a `nextRecordsUrl` field and return `true` value for field `done`

Ref Link for Issue:  https://blog.bessereau.eu/assets/pdfs/api_rest.pdf

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have made corresponding change to the default configuration files
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates #34057

### CODE REVIEW

**Comment by agithomas** [comment]
CreatedDate is changed to LogDate in the query.  Which one is most appropriate? Why ?
Is this change fixing any bug?

Kindly complete the internal code review.

**Comment by kush-elastic** [comment]
> CreatedDate is changed to LogDate in the query. Which one is most appropriate? Why ? Is this change fixing any bug?
> 

Currently, in the httpjson we are not able to use the pagination + split + chain call's , we had a similar Issue(https://github.com/elastic/beats/issues/33213) previously and we are in a chat with @ShourieG  who has worked on this issue previously(https://github.com/elastic/beats/pull/33664), He is currently looking into the issue, we have provided the required configurations to dig deep into this issue.

Issue
We are not getting the response for the second page.

Solution 
-> We removed the split functionality which was used to set cursor value of (CreatedDate) from the first response so that we are able to receive the response from the second page.

-> We changed the cursor value to more reliable data that we receive in the second call(chain call) TIMESTAMP_DERIVED which is related to the value between (LogDate  and LogDate + 1h) and has no relation with CreatedDate.

-> We have tried data collection without split functionality it's working fine as it was used for just setting the cursor value which is now changed to TIMESTAMP_DERIVED. Also, we have updated the query to LogDate as it is more related to TIMESTAMP_DERIVED.

Also, we have used the similar logic in our salesforce integration as well.
JFYI:

- LogDate tracks usage activity of a log file for a 24-hour period, from 12:00 a.m. to 11:59 p.m.
- CreatedDate tracks when the log file was generated.

> Kindly complete the internal code review.

This PR is internally reviewed but as discussed [here](https://github.com/elastic/integrations/pull/4866#issuecomment-1360915712) we don't have any other account as of now to provide approval, so consider this PR approved, Harnish can comment the same.

**Comment by harnish-elastic** [comment]
LGTM! Approved.

File: x-pack/filebeat/module/salesforce/logout-rest/config/logout-rest.yml [hunk 1]
```diff
@@ -29,18 +29,21 @@ request.proxy_url: {{ .proxy_url }}
 request.transforms:
   - set:
       target: url.params.q
-      value: "SELECT Id,CreatedDate,LogDate,LogFile FROM EventLogFile WHERE Interval = 'Hourly' AND EventType = 'Logout' AND CreatedDate > [[.cursor.last_published_logout]] ORDER BY CreatedDate ASC NULLS FIRST"
-      default: "SELECT Id,CreatedDate,LogDate,LogFile FROM EventLogFile WHERE Interval = 'Hourly' AND EventType = 'Logout' ORDER BY CreatedDate ASC NULLS FIRST"
-response.split:
-  target: body.records
+      value: "SELECT Id,CreatedDate,LogDate,LogFile FROM EventLogFile WHERE Interval = 'Hourly' AND EventType = 'Logout' AND LogDate > [[.cursor.last_published_logout]] ORDER BY CreatedDate ASC NULLS FIRST"
```
Comment (agithomas) [inline x-pack/filebeat/module/salesforce/logout-rest/config/logout-rest.yml:32]: The orderby clause must be `LogDate` . 
Please make the similar correction in 

https://github.com/elastic/integrations/blob/main/packages/salesforce/data_stream/logout_rest/agent/stream/httpjson.yml.hbs

**Review by agithomas — APPROVED** [review]
Approving. 

We need to see the outcome of testcases run to resolve the SDH issue.  According to the outcome,  the queries may need to be modified.

**Pull request merged**